pkg/provisioner/cloudprovisioner.go

```diff
@@ -62,11 +62,12 @@ type listVolumeStatus struct {
 }
 
 func NewCloudProvisioner(
+	userAgent string,
```

Comment:
Why prepend `userAgent` string to the parameter list instead of keeping it in the same order as `GetAzureCloudProvider` (i.e. after `cloudConfigSecretNamespace`)?